sklearn/base.py

```diff
@@ -1124,9 +1124,12 @@ def decorator(fit_method):
         @functools.wraps(fit_method)
         def wrapper(estimator, *args, **kwargs):
             global_skip_validation = get_config()["skip_parameter_validation"]
+
+            # we don't want to validate again for each call to partial_fit
             partial_fit_and_fitted = (
                 fit_method.__name__ == "partial_fit" and _is_fitted(estimator)
             )
+
             if not global_skip_validation and not partial_fit_and_fitted:
                 estimator._validate_params()
```

Comment:
How about computing `partial_fit_and_fitted` inside `if not global_skip_validation:`? I don't know if `_is_fitted(estimator)` is always fast, because may check all parameter names of the estimator or call `estimator.__sklearn_is_fitted__()` if it is present. It might be a good idea to avoid it if not necessary.